qtfred/src/mission/dialogs/ReinforcementsEditorDialogModel.cpp

```diff
@@ -0,0 +1,324 @@
+		memset( Reinforcements[i].yes_messages, 0, MAX_REINFORCEMENT_MESSAGES * NAME_LENGTH );
+
+		i++;
+	}
+
+	_shipWingPool.clear();
+	_numberLineEditUpdateRequired = false;
+	_listUpdateRequired = false;
+	_selectedReinforcementIndices.clear();
+
+	return true;
+}
+
+void ReinforcementsDialogModel::reject() 
+{
+	_shipWingPool.clear();
+	_reinforcementList.clear();
+	_selectedReinforcementIndices.clear();
+	_numberLineEditUpdateRequired = false;
+	_listUpdateRequired = false;
+}
+
+bool ReinforcementsDialogModel::numberLineEditUpdateRequired()
+{
+	return _numberLineEditUpdateRequired;
+}
+
+bool ReinforcementsDialogModel::listUpdateRequired()
+{
+	return _listUpdateRequired;
+}
+
+void ReinforcementsDialogModel::addToReinforcements(const SCP_vector<SCP_string>& namesIn)
+{
+	for (auto& newReinforcement : namesIn) {
+
+		for (auto candidate = _shipWingPool.begin(); candidate != _shipWingPool.end(); candidate++) {
+			if (*candidate == newReinforcement) {
+				// the default use number is 1, the default delay is zero
+				_reinforcementList.emplace_back(*candidate, 1, 0);
+				_shipWingPool.erase(candidate); // not efficient, but order matters.
+				break;
+			}
+		}
+	}
+
+	while (_reinforcementList.size() > MAX_REINFORCEMENTS) {
+		_shipWingPool.push_back(std::get<0>(_reinforcementList.back()));
+		_reinforcementList.pop_back();
+	}
+
+	_listUpdateRequired = true;
+	_numberLineEditUpdateRequired = true;
+	modelChanged();
+}
+
+void ReinforcementsDialogModel::removeFromReinforcements(const SCP_vector<SCP_string>& namesIn) 
+{
+	SCP_vector<SCP_string> removalList;
+
+	for (auto& removal : namesIn) {
+		for (auto candidate = _reinforcementList.begin(); candidate != _reinforcementList.end(); candidate++) {
+			if (std::get<0>(*candidate) == removal) {
+				removalList.push_back(std::get<0>(*candidate));
+				_shipWingPool.push_back(removal);
+				_reinforcementList.erase(candidate); // not efficient, but order matters.
+				break;
+			}
+		}
+	}
+
+	_selectedReinforcements.clear();
+	updateSelectedIndices();
+
+	_listUpdateRequired = true;
+	_numberLineEditUpdateRequired = true;
+	modelChanged();
+}
+
+// remember to call this and getReinforcementList together
+SCP_vector<SCP_string> ReinforcementsDialogModel::getShipPoolList()
+{
+	return _shipWingPool;
+}
+
+// remember to call this and getShipPoolList together
+SCP_vector<SCP_string> ReinforcementsDialogModel::getReinforcementList()
+{
+	_listUpdateRequired = false;
+
+	SCP_vector<SCP_string> list;
+
+	for (auto& currentReinforcement : _reinforcementList)
+	{
+		list.push_back(std::get<0>(currentReinforcement));
+	}
+
+	return list;
+}
+
+int ReinforcementsDialogModel::getUseCount() 
+{
+	if (_selectedReinforcementIndices.empty()) {
+		return -2;
+	}
+
+	int previous = -1, current = -1;
+
+	for (auto& item : _selectedReinforcementIndices) {
+		current = std::get<1>(_reinforcementList.at(item));
+
+		if (previous == -1 || current == previous) {
+			previous = current;
+			continue;
```

Comment:
This particular `continue` can be removed as it has no effect.

Reply:
Done.